### ISSUE TITLE
validation fails with 3.7.1. when referencing imported type

### ISSUE THREAD

**Issue by schiller-manuel** [fabien0102/ts-to-zod#207]
After #205 was merged, validation still fails in v3.7.1 for the following reproducer:

https://github.com/fabien0102/ts-to-zod/commit/502b7c08233c33c60f09c298d76d0a756a170b7b

```shell
✖ Validating generated types
 ›   Error: 'barSchema' is not compatible with 'Bar':
 ›   Argument of type '{ [x: string]: any; hello?: { foo: string; } | undefined; desc?: any; }' is not assignable to parameter of type 'Bar'.
 ›     Types of property 'hello' are incompatible.
 ›       Type '{ foo: string; } | undefined' is not assignable to type 'Foo'.
 ›         Type 'undefined' is not assignable to type 'Foo'.
```

@tvillaren please have a look

**Comment by tvillaren** [comment]
Yes, it's the case that we're trying to had handling for (not the one originally mentioned in #203)

Just finished #204 to cover this case.

**Comment by schiller-manuel** [comment]
I must admit I lost track on what was left to solve... I thought #205 would solve all validation errors, but at the expense of "type-safety" since all "internal" and "external" imports where treated as `any`.

But in the above example, the error message does not even concern an imported type:

```Types of property 'hello' are incompatible.```

```tsx
import { Description } from "./description";

export interface Foo {
  foo: string;
}

export interface Bar {
  hello: Foo; // <<< this is not imported
  desc: Description;
}
````

**Comment by tvillaren** [comment]
Indeed, that's weird as it passed the CI...

Can you try against my latest push (https://github.com/fabien0102/ts-to-zod/pull/204/commits/946d7c8908d9b5423cb0e2e4eea30a6f4b37c3ff)? It does pass on my side, provided you generate the types in the right order (or run `yarn gen:all` 3 times)

**Comment by tvillaren** [comment]
With regards to the issue on `"hello"` it's a property ordering thing: if you switch `hello` and `description`: 

```ts

export interface Bar {
  hello: Foo; // <<< this is not imported
  desc: Description;
}
```

you'll have 
```
 Error: 'barSchema' is not compatible with 'Bar':
 ›   Argument of type '{ [x: string]: any; desc?: any; hello?: { foo: string; } | undefined; }' is not assignable to parameter of 
 ›   type 'Bar'.
 ›     Property 'desc' is optional in type '{ [x: string]: any; desc?: any; hello?: { foo: string; } | undefined; }' but required 
 ›   in type 'Bar'.
```

**Comment by tvillaren** [comment]
This was fixed by #207 and released in 3.7.3
